0000-pre-shared-key-v2.md

```diff
@@ -0,0 +1,324 @@
+
+### Payment Chunks
+
+#### Payment Chunk Request
+
+The following is encrypted, as defined [above](#encryption), and set as the ILP payment data on a payment from the sender to the receiver.
+
+```
+(Numbers represent bytes)
+
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|T|           Payment ID          |  Seq  | Payment Total | Chu-|
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|nk Min   |             Application Data                        |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                       Application Data                        |
+\                                                               \
+\                                                               \
+|                       Application Data                      |E|
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                          Junk Data                            |
+\                                                               \
+\                                                               \
+|                          Junk Data                            |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+```
+
+| Field | Type | Description |
+|---|---|---|
+| Type (T) | 1-Byte UInt (UInt8) | `2` to indicate a payment chunk, `3` to indicate it is the last chunk for the payment. Receivers SHOULD NOT fulfill additional chunks for the given payment after the last chunk has been received |
+| Payment ID | 16-Byte UInt | Unique ID of this payment |
+| Sequence | 4-Byte UInt (UInt32) | Zero-indexed chunk sequence number. Receivers SHOULD NOT fulfill two chunks with the same sequence for a single payment. Receivers SHOULD use the sequence to order chunks in case application data is split across multiple chunks, but chunks do not necessarily need to be delivered in order |
+| Payment Total | 8-Byte UInt (UInt64) | The total amount the payment is intended to deliver. A value of `0` in a request indicates that the sender will determine when to stop sending chunks |
+| Chunk Minimum | 8-Byte UInt (UInt64) | Minimum amount the receiver should receive for this chunk in order to fulfill it. A value of `0` in a request indicates that any amount should be accepted |
+| Application Data | [OER Variable-Length Octet String](http://www.oss.com/asn1/resources/books-whitepapers-pubs/Overview%20of%20OER.pdf) | User data carried along with the payment |
+| Extensions (E) | 1-Byte UInt (UInt8) | Always `0`. Reserved for future use to indicate the presence of extension fields |
+| Junk Data | 0-? Bytes | _(Optional)_ Extra data included to obscure what the ILP payment is for. Receivers MUST ignore this data. Senders SHOULD include either the maximum or a random number of empty (0) bytes |
+
+**TODO:** should we add a source address for refunds? (that should only be necessary in extreme cases)
```

Comment:
I think that refunds are an uncommon enough case that psk shouldn't account for them.